controllers/notes.js

```diff
@@ -1,9 +1,38 @@
-exports.getNoteById = (req, res, next) => {
-        res.status(200).json({
-        success: true,
-         msg: `Note with id: ${req.params.id}`
-    });
+/*These controllers, when the endpoints are ChannelSplitterNode, return JSON, or wahtever */
+
+//this file contains the functions of the app
+
+exports.updateNote = (req, res, next) => {
+  Note.findByIdandUpdate(req.params.ide, req.body, {runValidators:true }, ()=>{
+    if (note) {
+      res.render('views/')
+    } else {
+      console.log('This note doesn\'t exist')
     }
+  });
+  //we are oging to have an arr of elements
+  //check composition bc there could be issues updating it since tis an arr
+  //if you are in template a, redirect to template b
+  res.redirect('user/notes');
+
+}
+
+exports.getNote = (req, res, next) => {
+  Note.findByIde(req.params.id)
+  .then(
+    note => res.render('views/edit.hbs', {note:note})
+  ).catch(
+    //next(new Error("custom error")) //and let the middleware handler, the error
+    err => res.render('views/edit.hbs', {error:err})
+  );
+
+}
+
+exports.getNoteById = (req, res, next) => {
```

Comment:
This can be delete because is the same as above (you could change the name thought)